src/openrct2-ui/windows/Ride.cpp

```diff
@@ -375,28 +375,28 @@ static Widget window_ride_income_widgets[] = {
 };
 
 // 0x009AE9C8
-static Widget window_ride_customer_widgets[] = {
+static Widget _customerWidgets[] = {
     MAIN_RIDE_WIDGETS,
     MakeWidget({289,  54}, {24, 24}, WindowWidgetType::FlatBtn, WindowColour::Secondary, ImageId(SPR_SHOW_GUESTS_THOUGHTS_ABOUT_THIS_RIDE_ATTRACTION), STR_SHOW_GUESTS_THOUGHTS_ABOUT_THIS_RIDE_ATTRACTION_TIP),
     MakeWidget({289,  78}, {24, 24}, WindowWidgetType::FlatBtn, WindowColour::Secondary, ImageId(SPR_SHOW_GUESTS_ON_THIS_RIDE_ATTRACTION),             STR_SHOW_GUESTS_ON_THIS_RIDE_ATTRACTION_TIP            ),
     MakeWidget({289, 102}, {24, 24}, WindowWidgetType::FlatBtn, WindowColour::Secondary, ImageId(SPR_SHOW_GUESTS_QUEUING_FOR_THIS_RIDE_ATTRACTION),    STR_SHOW_GUESTS_QUEUING_FOR_THIS_RIDE_ATTRACTION_TIP   ),
     WIDGETS_END,
 };
 
-static Widget *window_ride_page_widgets[] = {
-    window_ride_main_widgets,
-    window_ride_vehicle_widgets,
-    window_ride_operating_widgets,
-    window_ride_maintenance_widgets,
-    window_ride_colour_widgets,
-    window_ride_music_widgets,
-    window_ride_measurements_widgets,
-    window_ride_graphs_widgets,
-    window_ride_income_widgets,
-    window_ride_customer_widgets,
+static const std::array<Widget*, WINDOW_RIDE_PAGE_COUNT> PageWidgets = {
+    _mainWidgets,
+    _vehicleWidgets,
+    _operatingWidgets,
+    _maintenanceWidgets,
+    _colourWidgets,
+    _musicWidgets,
+    _measurementWidgets,
+    _graphsWidgets,
+    _incomeWidgets,
+    _customerWidgets,
 };
 
-static constexpr const uint64_t window_ride_page_hold_down_widgets[] = {
+static constexpr std::array<uint64_t, WINDOW_RIDE_PAGE_COUNT> PageHoldDownWidgets = {
```

Comment:
Please add a `static_assert(PageHoldDownWidgets.size() == WINDOW_RIDE_PAGE_COUNT);` to ensure this array will be filled properly in case new tabs are added. Otherwise this won't result in a compile time error when WINDOW_RIDE_PAGE_COUNT increases. Same goes for the other arrays.

Reply:
That won't work as I've specified the array size. You can use the template deduction guide but only for both parameters so with that we wouldn't be able to force it to uint64_t and it would deduce to int32_t. When we switch to c++20 we can use the alternative deduction guide which lets you specify the type and not the size.

So yeah i could readd the static asserts but they will never be false. I could add the asserts and remove the template parameters but then it will truncate to int32. I could add a bunch of casts to force uint64. None of the options are very good (until c++20).

Reply:
Hmm, it's what we've done in some windows already. The casts don't seem necessary for most arrays, and in this case the values are already uint64_t, right? Only the zeros will have to be changed to `0uLL` then. For non-trivial types it's easier:

https://github.com/OpenRCT2/OpenRCT2/blob/4a3a1e4e7fa442cda53f9c3d86b7d8e97989274e/src/openrct2-ui/windows/Guest.cpp#L141-L150

I agree that specifying the type without and no size would be ideal, but we'll have to wait for C++20 then.

Reply:
Okay I've changed it back to readd the static asserts. I've had to add a bunch of static_casts to make the StringId tables use the correct types